charts/gha-runner-scale-set/templates/autoscalingrunnerset.yaml

```diff
@@ -28,6 +28,7 @@ metadata:
     {{- if and (ne $containerMode.type "kubernetes") (not .Values.template.spec.serviceAccountName) }}
     actions.github.com/cleanup-no-permission-service-account-name: {{ include "gha-runner-scale-set.noPermissionServiceAccountName" . }}
     {{- end }}
+    argocd.argoproj.io/sync-wave: "1"
```

Comment:
Just cross-posting my [comment](https://github.com/actions/actions-runner-controller/issues/3440#issuecomment-2073591886) from #3440 

> I do use `sync-waves` on the top level resources to ensure the `gha-runner-scale-set-controller` syncs first and the `gha-runner-scale-set-figure-linux-standard` syncs second:

  ```
  metadata:
  name: gha-runner-scale-set-controller
  namespace: argocd
  annotations:
    argocd.argoproj.io/sync-wave: "1" # before scale-set

  ---

  metadata:
  name: gha-runner-scale-set-figure-linux-standard
  namespace: argocd
  annotations:
    argocd.argoproj.io/sync-wave: "2" # after controller
  ```

  > Looking at your PR you set `argocd.argoproj.io/sync-wave: "1"` - would that conflict in my case with my controller having the same sync wave? Would it makes sense to go negative on the sync-wave for the autoscalingrunnerset to always ensure that runs first?

Reply:
https://github.com/actions/actions-runner-controller/issues/3440#issuecomment-2075059032